### PR TITLE
Filesystem changes

### PR DESCRIPTION
**Pull request by Smolevich**
Add additional conditions for functions directories,  allFiles, files, delete, put, getRequire, get

### CODE REVIEW

**Comment by TBlindaruk** [comment]
I think it will be `Breaking changes`.

**Comment by sisve** [comment]
1. Why did you change from specific exception types to ErrorException? This would be a breaking change.
2. Every failed test could be a breaking change, and there are  loads of failing tests.

It sounds like this should be explained, fixed and target the master branch if it has breaking changes.

File: src/Illuminate/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -30,15 +29,19 @@ public function exists($path)
...
-            return $lock ? $this->sharedGet($path) : file_get_contents($path);
+            if ($this->isReadable($path)) {
+                return $lock ? $this->sharedGet($path) : file_get_contents($path);
+            }
+
+            throw new ErrorException("Failed to read from {$path}");
         }
 
-        throw new FileNotFoundException("File does not exist at path {$path}");
+        throw new ErrorException("File does not exist at path {$path}");
```
Comment (mputkowski) [inline src/Illuminate/Filesystem/Filesystem.php]: What's the reason to replace that?

**Comment by taylorotwell** [comment]
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!

**Pull request closed without merging**